frontend/src/util/err.ts

```diff
@@ -0,0 +1,40 @@
+// Make sure that the parameter is `never`. This is particularly useful for
+// enum-like types where you want to make sure you handled all cases. For
+// example:
+//
+// ```
+// type Foo = "anna" | "bob";
+// const x = "anna" as Foo;
+//
+// switch x {
+//     case "anna": ...; break;
+//     case "bob": ...; break;
+//     default: assertNever(x);
+// }
+// ```
+//
+// If you add another variant to that type, the switch statement will fail to
+// compile. This is what we want in most cases! That way, we are notified of all
+// places which we might need to change after adding a new variant.
+export const assertNever = (n: never): never => n;
```

Comment:
Should this `throw`? TS's type system is unsound, and deliberately offers ways to circumvent it, which we already use (see `util/index.js`). If we or TS ever get it wrong, just returning the argument here might lead to surprising and potentially hard to debug problems. I think I would prefer an exception as a second line of defense, here.

The counterargument would be that functions throwing exceptions are probably less optimizable for the JS engine, and I doubt the TS compiler will already optimize this out, precisely because of what I said above.

Reply:
I added throwing an exception here. I agree a second line of defense is good here.